### DISCUSSION TITLE
How to use an IconButton as the trigger for Popover?

### DISCUSSION THREAD

**Discussion by janhassel** [carbon-design-system/carbon#13836] [Help] [answered]
I'm running into the issue that I wanna use an `IconButton` (or another icon-only element) as the trigger for a Popover. To ensure the user understands the action of the icon, a tooltip should appear.
But because the `IconButton` (as well as `Tooltip`) is using `Popover` under the hood a nested Popover is created and causes some issues when rendering, including:
- spacing issues
- the tooltip popover stays open as long as the actual content popover is open

Is there a better/intended way to achieve this behaviour? The "Popover/Tab Tip" story doesn't use a tooltip on the trigger button.

Here is a quick reproduction: https://stackblitz.com/edit/github-yaemgv?file=src%2FApp.jsx

<img width="525" alt="image" src="https://github.com/carbon-design-system/carbon/assets/28265588/af9e6948-b4e2-4428-b98a-95d45b2ed889">

**Comment by tay1orjones** [comment]
@janhassel I think this will be resolved by https://github.com/carbon-design-system/carbon/pull/13750

**Reply by janhassel** [reply]
I just tried it with `@carbon/react@1.30.0` and the spacing issue unfortunately remains. It does close now properly though.

https://stackblitz.com/edit/github-yaemgv-vppy9h

**Reply by tay1orjones** [reply, marked as answer]
@janhassel Sorry for the delay here, looks like a follow up was issued https://github.com/carbon-design-system/carbon/pull/14200

I'm not seeing the spacing issue anymore 

![image](https://github.com/carbon-design-system/carbon/assets/3360588/927cad88-e8ea-40c5-9eba-38827e3c4ce6)

https://stackblitz.com/edit/github-yaemgv-jd41kh?file=package.json

**Reply by janhassel** [reply]
Works now, thank you!
